### PR TITLE
New graph operations cheat sheet

### PR DESCRIPTION
**Pull request by navneetzz**
**What does your Instant Answer do?**
My Instant answer provide reference to Big-O complexities of graph operations.

**What is the data source for your Instant Answer? (Provide a link if possible)**
[Wikipedia](https://en.wikipedia.org/wiki/Graph_theory)

**Why did you choose this data source?**
Because it's detailed reference to graph operations.

**Are there any other alternative (better) data sources?**
might be

**Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)**
Coders and programmers

**Where did you hear about DuckDuckHack? (For first time contributors)**
Social media sites.

graph-operations

---

---
https://duck.co/ia/view/graph_operations_cheat_sheet

### CODE REVIEW

**Comment by daxtheduck** [comment]
## [Graph Operations Cheat Sheet](https://duck.co/ia/view/graph_operations_cheat_sheet)

**Description**: A quick reference to Big-O complexities of graph operations

**Example Query**: [graph operations cheat sheet](https://beta.duckduckgo.com/?q=graph operations cheat sheet), [](https://beta.duckduckgo.com/?q=)

**Tab Name**: Answer

**Source**: Wikipedia

*These are the important fields from the IA page.  Please check these for errors or missing information and update the [IA page](https://duck.co/ia/view/graph_operations_cheat_sheet)*

---
*This is an automated message which will be updated as changes are made to the [IA page](https://duck.co/ia/view/graph_operations_cheat_sheet)*

File: share/goodie/cheat_sheets/json/graph-operations.json [hunk 1]
```diff
@@ -0,0 +1,97 @@
+{
+    "id": "graph_operations_cheat_sheet",
+    "name": "Graph Operations",
+    "description": "A quick reference to Big-O complexities of graph operations",
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/graph-operations.json]: How about... `Summary of Big-O run-time complexities for common graph operations` or similar.

File: share/goodie/cheat_sheets/json/graph-operations.json [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+            "key": "Remove Vertex"
+        }, {
+            "val": "O(|E|)",
+            "key": "Remove Edge"
+        }, {
+            "val": "O(|E|)",
+            "key": "Query"
+        }],
+        "Adjacency matrix": [{
+            "val": "O(|V|^2)",
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/graph-operations.json]: We can use Unicode symbols, so lets replace all the `^2` with `²`.

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=graph operations cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/graph_operations_cheat_sheet_mobile.png)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=graph operations cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/graph_operations_cheat_sheet_mobile.png)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=graph operations cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/graph_operations_cheat_sheet_mobile.png)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=graph operations cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/graph_operations_cheat_sheet_mobile.png)

**Comment by GuiltyDolphin** [comment]
@moollaza Looks good to me!

Though same comment as for #2243.

**Comment by moollaza** [comment]
LGTM :+1:

**Comment by moollaza** [comment]
@navneetzz FWIW you don't need to use the PR Template any more

**Pull request merged**